### PR TITLE
Flyout - set correct size in Full placement mode

### LINKED ISSUE #2009: [Android][iOS] Flyout does not appear full screen
## Current behavior

Opening a flyout with a full-screen adjusted presenter does not fill the entire window area.

## Expected behavior

Opening a flyout with a full-screen adjusted presenter does fill the entire window area.

## How to reproduce it (as minimally and precisely as possible)

- Run the following project: [FullScreenFlyoutIssue.zip](https://github.com/unoplatform/uno/files/3795538/FullScreenFlyoutIssue.zip)
- Click on the button, notice the popup appears centered, with a dismiss area incorrectly placed.

image

## Environment

Package Version(s): 2.0.512-dev.3668

Affected platform(s):

- [x] iOS
- [x] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

Visual Studio:

- [ ] 2017 (version: )
- [ ] 2019 (version: )
- [ ] for Mac (version: )

Relevant plugins:

- [ ] Resharper (version: )

### PR DESCRIPTION
**Pull request by davidjohnoliver**
GitHub Issue (If applicable): fixes #2009

 - FlyoutPresenter takes up all available size up to MaxSize when placement mode is Full
 - fix exception when MaxHeight/MaxWidth=NaN

## PR Type

What kind of change does this PR introduce?

- Bugfix

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/UIElement.net.cs [hunk 1]
```diff
@@ -48,5 +48,22 @@ internal Windows.Foundation.Point GetPosition(Point position, global::Windows.UI
...
+			return new MatrixTransform
+			{
+				Matrix = new Matrix(
+					m11: 1,
+					m12: 0,
+					m21: 0,
+					m22: 1,
+					offsetX: 0, //TODO
+					offsetY: 0  //TODO
+				)
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/UIElement.net.cs]: `TODO` are not just for `Offset<X|Y>` ;)

Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/UIElement.net.cs]: Fair enough, but they're TODO in that case for all platforms, not just unit tests :p

Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/UIElement.net.cs]: No, the `TransformToVisual` is implemented now on iOS, Android and WASM 🤔

Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/UIElement.net.cs]: Ok, I guess I can't see where they're setting the non-translate matrix components then.

https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/UIElement.Android.cs#L71
https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/UIElement.iOS.cs#L142
https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/UIElement.wasm.cs#L618

File: src/Uno.UI/UI/Xaml/FrameworkElement.Interface.net.cs [hunk 2]
```diff
@@ -114,5 +114,171 @@ protected virtual void OnBackgroundChanged(DependencyPropertyChangedEventArgs e)
...
+			);
+
+		public HorizontalAlignment HorizontalAlignment
+		{
+			get { return (HorizontalAlignment)this.GetValue(HorizontalAlignmentProperty); }
+			set { this.SetValue(HorizontalAlignmentProperty, value); }
+		}
+		#endregion
+
+		#region HorizontalAlignment Dependency Property
```
Comment (XiaotianNetlift) [inline src/Uno.UI/UI/Xaml/FrameworkElement.Interface.net.cs]: Vertical

File: src/Uno.UI/UI/Xaml/UIElement.net.cs (hunk 1 above)
Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/UIElement.net.cs]: Hummm right ... I thought the `TransformBounds`, but it's on the `GeneralTransform` (https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/Media/GeneralTransform.cs#L31) ... so it means that my next bug will be harder to fix 😥

File: src/Uno.UI/UI/Xaml/FrameworkElement.Interface.net.cs (hunk 2 above)
Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/FrameworkElement.Interface.net.cs]: Good catch!

**Pull request merged**